spec/main-process/atom-application.test.js

```diff
@@ -368,6 +370,23 @@ describe('AtomApplication', function () {
     })
   })
 
+  describe('before quitting', function () {
+    it('waits until all the windows have saved their state and then quits', async function () {
```

Comment:
These are main process tests and use Mocha. Eventually we want to use Mocha for our core render process tests too but its not a priority right now. You can use Mocha today for packages via [atom-mocha-test-runner](https://github.com/binarymuse/atom-mocha-test-runner].